lib/src/model.dart

```diff
@@ -1119,7 +1119,15 @@ abstract class SourceCodeMixin {
         break;
       }
     }
-    return contents.substring(node.offset - (node.offset - i), node.end);
+
+    // Trim the common indent from the source snippet.
+    String source =
+        contents.substring(node.offset - (node.offset - i), node.end);
+    String remainer = source.trimLeft();
+    String indent = source.substring(0, source.length - remainer.length);
+    return source.split('\n').map((line) {
```

Comment:
Yup, it'll split `\r\n` lines into things that end in `\r`, and then we'll add the `\n` back on with the join(). We could also right-trim the strings...